BeeSwift/Gallery/GalleryViewController.swift

```diff
@@ -504,3 +524,17 @@ class GalleryViewController: UIViewController, UICollectionViewDelegateFlowLayou
         self.fetchGoals()
     }
 }
+
+extension GalleryViewController: NSFetchedResultsControllerDelegate {
+    func controllerDidChangeContent(_ controller: NSFetchedResultsController<NSFetchRequestResult>) {
+        self.applySnapshot()
```

Comment:
[Consider]
https://developer.apple.com/videos/play/wwdc2019/230/?time=890 suggests that we should be able to get a snapshot from the NSFetchedResultsController and apply it directly to the `DiffableDataSource`.

<img width="721" alt="image" src="https://github.com/user-attachments/assets/0f59a771-30c8-4f0d-a624-dfe81eb00ea6" />

https://stackoverflow.com/a/58475482 suggests that this can work, although you need to use `UITableViewDiffableDataSource<String,NSManagedObjectID>`